.github/workflows/build.yml

```diff
@@ -0,0 +1,43 @@
+# The present workflow was made based on the following references:
+# - https://github.com/actions/cache/blob/main/examples.md#haskell---cabal
+# - https://github.com/haskell/time/blob/master/.github/workflows/ci.yml
+# - https://github.com/stackbuilders/stache/blob/master/.github/workflows/ci.yaml
+# - https://markkarpov.com/post/github-actions-for-haskell-ci.html
+---
+name: Build
+
+on: push
+
+concurrency:
+  group: build-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  haskell:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        ghc: ['9.0', '8.10', '8.6.1', '8.4.3', '8.2']
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install dependencies (Linux)
```

Comment:
@cptrodolfox I don't think we need to install `zsh` in the context of this project

Reply:
@sestrella Thanks I totally missed the installation of zsh. Will remove it.